packages/terra-image/src/Image.jsx

```diff
@@ -47,36 +51,106 @@ const defaultProps = {
   alt: ' ',
 };
 
-const Image = ({
-  src,
-  variant,
-  isFluid,
-  alt,
-  height,
-  width,
-  onLoad,
-  onError,
-  ...customProps
-}) => {
-  const classes = cx([
-    'image',
-    variant,
-    { fluid: isFluid },
-    customProps.className,
-  ]);
-
-  return (
-    <img
-      src={src}
-      alt={alt}
-      height={height}
-      width={width}
-      onLoad={onLoad}
-      onError={onError}
-      {...customProps}
-      className={classes}
-    />);
-};
+class Image extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { isLoading: true, isError: false };
+    const { src, variant, isFluid, alt, placeholder, height, width, onLoad, onError, ...customProps } = this.props;
+
+    this.classes = cx([
+      'image',
+      variant,
```

Comment:
The `variant`, `isFluid`, and `customProps.className` props aren't update safe. Since you're caching this set of classes in the constructor, you'd need to recreate it in componentWillReceiveProps in case those props have changed values.

It seems like it'd be nice to cache the classnames, but it might be more work than desirable. If you decide to keep this caching, I'd just store it in state directly. It'll make your componentWillReceiveProps implementation more complicated, but it'd be good to sync the classnames with the normal component refresh cycle.

Reply:
My vote is not to cache the classes personally. Do we get a huge performance gain by caching them?

Reply:
It would be safer and more straightforward to implement 'shouldComponentUpdate' to prevent unnecessary updates when props haven't changed (if we're really concerned about this component's performance).